app/routes/goal_routes.py

```diff
@@ -0,0 +1,105 @@
+from flask import Blueprint, request, make_response, jsonify, abort
+from flask.wrappers import Response
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+from datetime import date, datetime
+
+goals_bp = Blueprint('goals', __name__, url_prefix='/goals')
+
+# # Helper functions
+def valid_int(id):
+    try:
+        number = int(id)
+        return number
+    except:
+        response_body = 'Invalid Data'
+        abort(make_response(response_body,400))
+
+def get_goal_from_id(goal_id):
+    id = valid_int(goal_id)
+    selected_goal = Goal.query.filter_by(goal_id=goal_id).one_or_none()
+    # Goal not found
+    if selected_goal is None:
+        abort(make_response("Not Found", 404))
+    return selected_goal
+
+def valid_goal(request_body):
+    if "title" not in request_body:
+        abort(make_response({"details": "Invalid data"}, 400))
+# #
+
+# Create a goal
+@goals_bp.route("", methods=["POST"], strict_slashes=False)
+def create_goal():
+    request_body = request.get_json()
+    valid_goal(request_body)
```

Comment:
Great use of a helper function. You might consider also including the creation of the goal in `valid_goal` such that the function either aborts because of Invalid data or returns the goal instance.